kuma/tests/test_tag_injection.py

```diff
@@ -0,0 +1,155 @@
+# (C) Datadog, Inc. 2025-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from pathlib import Path
+from typing import Any, Callable, Dict  # noqa: F401
+
+import pytest
+from prometheus_client.core import CounterMetricFamily, Sample
+
+from datadog_checks.base.stubs.aggregator import AggregatorStub  # noqa: F401
+from datadog_checks.kuma import KumaCheck
+from datadog_checks.kuma.check import KumaOpenMetricsScraper
+
+
+@pytest.fixture
+def setup_check(dd_run_check, instance, mock_http_response):
+    """Fixture to set up and run the KumaCheck with a mocked HTTP response."""
+    mock_http_response(
+        file_path=Path(__file__).parent.absolute() / "fixtures" / "metrics" / "control_plane" / "metrics_code_class.txt"
+    )
+    check = KumaCheck('kuma', {}, [instance])
+    dd_run_check(check)
+    return check
+
+
+@pytest.mark.parametrize(
+    'code,expected_class',
+    [
+        pytest.param('200', '2XX', id='2xx-ok'),
+        pytest.param('302', '3XX', id='3xx-found'),
+        pytest.param('401', '4XX', id='4xx-unauthorized'),
+        pytest.param('500', '5XX', id='5xx-internal'),
+    ],
+)
+def test_code_class_injection_valid_codes(aggregator, code, expected_class, setup_check):
+    """Test that valid HTTP status codes get correct code_class tags"""
+    metric_name = 'kuma.api_server.http_requests_inflight'
+    aggregator.assert_metric(metric_name)
+
+    # Test that the specific code gets correct code_class - filter metrics with the code first
+    matching_metrics = [
+        metric_point for metric_point in aggregator.metrics(metric_name) if f'code:{code}' in metric_point.tags
+    ]
+
+    assert matching_metrics, f"No metric found for {metric_name} with tag code:{code}"
+
+    for metric_point in matching_metrics:
+        assert (
+            f'code_class:{expected_class}' in metric_point.tags
+        ), f"Expected code_class:{expected_class} for code:{code}, got tags: {metric_point.tags}"
```

Comment:
What would you think about avoid doing multiple asserts and instead build an error message and one single assert? I normally prefer to have a single assert that reports the error instead of multiple asserts that can cause repeated iterations to fix the test. If we have 4 asserts and the second one breaks the test fails. We can fix that but at this point we have no information about the last 2, which could also fail and we need to keep iterating.

There is a library called `pytest-check` which allows you do do this. Multiple asserts that report at the end. But I don't think we want to include it as a dependency and is rather small.

I would modify this test to do the following

```suggestion
    errors = []
    for metric_point in matching_metrics:
        expected_tag = f'code_class:{expected_class}'
        if expected_tag not in metric_point.tags:
            errors.append(
                f"Expected '{expected_tag}' for code:{code}. Got tags: {metric_point.tags}"
            )

    assert not errors, "Found metric tag mismatches:\n" + "\n".join(errors)
```

WDYT?

Reply:
What do you mean with the debugger break on assertions? What is it that we are loosing from moving to a single assert? Not sure I understood the line numbers issue either.

We will still get the error if any tag is missing, you just get a full report instead of one failure at a time and you can always add a break point in every iteration to understand why it is failing while debugging.

We could add `pytest-check` as a dependency but I am not sure I feel confortable bringing that small dependency into an integration since I am not sure how maintained it would be.

Reply:
> What do you mean with the debugger break on assertions? What is it that we are loosing from moving to a single assert? Not sure I understood the line numbers issue either.

If you enable catching user uncaught exceptions in the debugger you can get it to break where an assertion fails. However if the assertion doesn't happen in the location of the actual failure, but in the line where the single assertion for all previous errors happens, then the break will be less useful. Hopefully `pytest-check` will support that case but I'm not sure. Same situation for the line number. Anyway it's not a big deal here. 

> We could add pytest-check as a dependency but I am not sure I feel confortable bringing that small dependency into an integration since I am not sure how maintained it would be.

Oh, I meant adding it as a dependency only in the testing environment, for example by including it with ddev_checks_dev (similar to your other comment proposing implementing it there). It looks like it has a number of useful features! If we include them maybe it starts being big enough to be [dependency-worthy](https://github.com/okken/pytest-check/tree/main/src/pytest_check)?

We can think about this in another PR! I'll use the suggested strategy in the meantime!